Sources/SwiftSyntaxBuilder/BuildableNodes.swift.gyb

```diff
@@ -108,7 +108,15 @@ public struct ${type.buildable()}: ${base_type.buildable()}, ${type.expressible_
   /// Conformance to `${base_type.buildable()}`.
   public func build${base_type.base_name()}(format: Format, leadingTrivia: Trivia? = nil) -> ${base_type.syntax()} {
     let result = build${type.base_name()}(format: format, leadingTrivia: leadingTrivia)
-    return ${base_type.syntax()}(result).withLeadingTrivia(itemLeadingTrivia ?? .zero)
+    
+      
+     if let itemLeadingTrivia = itemLeadingTrivia
+    {
+      let formattedLeadingTrivia = [itemLeadingTrivia, format._makeIndent()].reduce(.zero, +)
```

Comment:
Thanks, @ahoppen for pointing me out to 

> If you now stick that ImportDecl into a SourceFile and add a comment to the SourceFile

I've solved it by doing the following, what do you think?، 
also, this should support comments indiation

Reply:
I think this will solve the indentation problem but not your problem with dropping `IMPORTANT COMMENT` as you described in the PR description.

I would try one of the following (haven’t tested myself):
```swift
if let itemLeadingTrivia = itemLeadingTrivia {
  leadingTrivia = itemLeadingTrivia + leadingTrivia // Alternative 1
  leadingTrivia = format._makeIndent() + itemLeadingTrivia + leadingTrivia // Alternative 2
}
let result = build${type.base_name()}(format: format, leadingTrivia: leadingTrivia)
```

and dropping the `withLeadingTrivia` down here.

Reply:
actually, I think it solved the dropping of the first Comment In `SourceFile {}`, for example:
the output is: 

```swift
// IMPORTANT COMMENT
import Foundation
// lineComment
// hello
/* This's the new block comment
with new lines support */
import UIKit
class SomeViewController{
    /** This's the first nested comment */
    let tableView: UITableView
    class SomeViewController{
        // this's a also a nested comment
        let mainNode: SCNNode
    }
}
```

for the following SourceFile 


```swift
  SourceFile {
    ImportDecl(path: "Foundation")
      .lineComment("IMPORTANT COMMENT")
    ImportDecl(path: "UIKit")
      .lineComment(
"""
lineComment
hello
"""
      )
      .blockComment("This's the new block comment\nwith new lines support")
    ClassDecl(classOrActorKeyword: .class, identifier: "SomeViewController", membersBuilder: {
      VariableDecl(.let, name: "tableView", type: "UITableView")
        .docBlockComment("This's the first nested comment")


      ClassDecl(classOrActorKeyword: .class,
                identifier: "SomeViewController",
                membersBuilder: {
        VariableDecl(.let, name: "mainNode", type: "SCNNode")
          .lineComment("this's a also a nested comment")
      })

    })
  }//.lineComment("This comment will override important comment if placed")
```

**(if a comment on SourceFile is placed, it will override the one on the first item, should we handle it in a special way, which adds both of the comments instead of overriding) ?**

is the current output enough for our needs?

please let me know if I'm missing something, also I will check your snippets and see how would they differ :-) 

Thank you for the review and for helping me.

Reply:
> **(if a comment on SourceFile is placed, it will override the one on the first item, should we handle it in a special way, which adds both of the comments instead of overriding) ?**

Yes, definitely. I don’t think we should be dropping comments if the user explicitly asked for them.

Could you add this test case to [Tests/SwiftSyntaxBuilderTest](https://github.com/apple/swift-syntax/tree/main/Tests/SwiftSyntaxBuilderTest)?